core/src/main/java/io/questdb/cutlass/http/HttpServer.java

```diff
@@ -102,16 +103,17 @@ public boolean run(int workerId, @NotNull RunStatus runStatus) {
 
     public static void addDefaultEndpoints(
             HttpServer server,
-            HttpServerConfiguration configuration,
+            ServerConfiguration serverConfiguration,
             CairoEngine cairoEngine,
             WorkerPool workerPool,
             int sharedWorkerCount,
             HttpRequestProcessorBuilder jsonQueryProcessorBuilder,
             HttpRequestProcessorBuilder ilpWriteProcessorBuilderV2
     ) {
+        final HttpServerConfiguration httpServerConfiguration = serverConfiguration.getHttpServerConfiguration();
+        final LineHttpProcessorConfiguration lineHttpProcessorConfiguration = httpServerConfiguration.getLineHttpProcessorConfiguration();
         // Disable ILP HTTP if the instance configured to be read-only for HTTP requests
-        if (configuration.getLineHttpProcessorConfiguration().isEnabled() &&
-                !configuration.getHttpContextConfiguration().readOnlySecurityContext()) {
+        if (httpServerConfiguration.isEnabled() && lineHttpProcessorConfiguration.isEnabled() && !httpServerConfiguration.getHttpContextConfiguration().readOnlySecurityContext()) {
```

Comment:
I added an extra condition here so that the failing tests would pass. I couldn't track down the specific commit that resulted in the failing tests tho

Reply:
@ngnic I think this change may have broken IPL.  I've been trying to setup 8.1.1, and have discovered that this change is the difference between the REST API endpoints working and not working. (/ping /write /api/v2/write)

In 7.3.9 where the original commit of #4034 was implement everything was working fine, when doing a curl -I http://localhost:9000/ping, I would get results that matched the documentation for the pong.

However, with commit and beyond it does not work. (at least for me).  Maybe there is a setting that is different from the documentation or a setting I missed, but based on the documentation, it should work, yet does not after this commit.